ydb/core/tx/schemeshard/schemeshard__table_stats.cpp

```diff
@@ -205,14 +231,17 @@ bool TTxStoreTableStats::PersistSingleStats(const TPathId& pathId,
     }
 
     TShardIdx shardIdx = Self->TabletIdToShardIdx[datashardId];
+    const auto& shardInfo = Self->ShardInfos[shardIdx];
+    auto subDomainInfo = Self->ResolveDomainInfo(pathId);
+    const auto channelsMapping = MapChannelsToStoragePoolKinds(subDomainInfo->EffectiveStoragePools(), shardInfo.BindedChannels);
```

Comment:
@snaury Is shardInfo.BindedChannels the most up-to-date information on the binding of the channels to the particular storage pool? Does it follow the Hive's balancing of the channels by storage pools based on usage?

Reply:
As far as I know it is the most up to date channel bindings, i.e. when altering table's column families we change `shardInfo.BindedChannels` when we prepare the operation, so before it is applied. However please note that older tablet generations may have stale info in memory.

Reply:
@snaury What do you mean by the "older tablet generations may have stale info in memory"? Does it mean that an old SchemeShard may have an outdated `shardInfo`? Is it going to be updated? Are there some messages that data shards send to SchemeShard to push updates of theirs infos?

Reply:
Discussed this privately with @snary and I believe the answer is yes, SchemeShard can have an outdated shardInfo, if it is on older generation tablet. However, such a tablet will not be able to change any object in the scheme, so the database will not be blocked for writing if the old SchemeShard decides that it has exceeded its quota based on the outdated information the old SchemeShard has on the channelsBinding. The worst thing that could happen is that the old SchemeShard will aggregate some table usage statistics that does not correspond to the real picture. However, this should not have any real impact on the cluster.